codex/validation.nim

```diff
@@ -1,35 +1,38 @@
 import std/sets
 import std/sequtils
 import pkg/chronos
+import pkg/questionable/results
+
+import ./contracts/validation
 import ./market
 import ./clock
 import ./logutils
 
 export market
 export sets
+export validation
 
 type
   Validation* = ref object
     slots: HashSet[SlotId]
-    maxSlots: int
     clock: Clock
     market: Market
     subscriptions: seq[Subscription]
     running: Future[void]
     periodicity: Periodicity
     proofTimeout: UInt256
+    validationParams: ValidationParams
 
 logScope:
   topics = "codex validator"
 
 proc new*(
-    _: type Validation,
-    clock: Clock,
-    market: Market,
-    maxSlots: int
+  _: type Validation,
+  clock: Clock,
+  market: Market,
+  validationParams: ValidationParams
```

Comment:
🟠 Just a suggestion, do you think possibly `config` or even `settings` might be better a more suitable name? That way, when called it would be, eg `validation.config.maxSlots`.

Reply:
I like `config`. Makes it more compact to read,